docs/windows.md

```diff
@@ -1,142 +1,94 @@
-
-## Download go-ipfs and fix Git authentication
+`go-ipfs` is built on Golang and thus depends on it for all building methods.  
+https://golang.org/doc/install  
+The `GOPATH` environment variable must be set as well.  
+https://golang.org/doc/code.html#GOPATH
 
-Use the following command to download go-ipfs source code:
-
-```
+## Choose the way you want to proceed
+`go-ipfs` utilizes `make` to automate builds and testing, but can be built without it using only `git` and `go`.  
+The `make` portions cover setting up the environment, "minimal" assumes you have a working version of `go` and `git` already.  
+[make→](#make)  
+[minimal→](#minimal)
+
+## Make
+There are currently 2 documented ways of building with `make` on Windows.  
+[MSYS2→](#msys2)  
+[Cygwin→](#cygwin)
+
+## MSYS2
+Install msys2 (http://www.msys2.org)  
+Run the following inside a normal `cmd` prompt (Not the MSYS2 prompt, we only need MSYS2's tools).  
+An explanation is below.
+```
+SET PATH=%PATH%;\msys64\usr\bin
+pacman --noconfirm -S  git make unzip
 go get -u github.com/ipfs/go-ipfs
+cd %GOPATH%\src\github.com\ipfs\go-ipfs
+make install
 ```
 
-The above command uses Git to download go-ipfs from its GitHub
-repository. If you get authentication problems with Git, you might
-want to take a look at
-https://help.github.com/articles/caching-your-github-password-in-git/
-and use the suggested solution:
+|      |      |
+| ---: | :--- |
+|`SET PATH=%PATH%;\msys64\usr\bin`         |Add msys2's bin to `PATH`; Defaults to: (\msys64\usr\bin)|
+|`pacman --noconfirm -S  git make unzip`   |Install `go-ipfs` build dependencies|
+|`go get -u github.com/ipfs/go-ipfs`       |Fetch / Update `go-ipfs` source|
+|`cd %GOPATH%\src\github.com\ipfs\go-ipfs` |Change to `go-ipfs` directory|
+|`make install`                            |Build and install to `%GOPATH%\bin\ipfs.exe`|
 
+You can permenantly append the MSYS2 bin to your `PATH` so that on subsequent builds you can simply run
 ```
-git config --global credential.helper wincred
+cd %GOPATH%\src\github.com\ipfs\go-ipfs
+make install
 ```
 
-## Choose the way you want to proceed
-
-Now there are two ways to download, install the dependencies and to
-build go-ipfs:
 
-1) There is the "Manual Way", where you don't need to install anymore
-software except the dependencies, but you have a number of commands to
-type.
-
-2) There is a way by installing 'make' through Cygwin and using it to
-do nearly everything. We call this way the "Cygwin Way". It may take
-much more time, because installing Cygwin can take a lot of time, but
-after that it might be easier as many procedures are just a 'make'
-command away.
-
-So use the next steps below that start with "Manual Way" if that's the
-way you want, otherwise scroll down a bit and use the "Cygwin Way"
-steps below.
+## Cygwin
+Install Cygwin (https://www.cygwin.com)  
+During the install, select the devel packages `git` & `make`, archive package `unzip`, and net package `curl`  
```

Comment:
Additionally, how should someone install these packages if they already have Cygwin? (If I recall, the answer is run the Cygwin setup again, but not sure if things have since improved.)

Reply:
You're right about needing to (download and) run the setup binary again, even if you already have it installed. I can see about adding some kind of wording like "Even if you already have Cygwin installed, you'll need to run the setup file again to add these packages"